cli/tests/test_split_command.rs

```diff
@@ -675,3 +680,199 @@ fn test_split_interactive_with_paths() {
     ◆  zzzzzzzz root() 00000000
     ");
 }
+
+// When a commit is split, the second commit produced by the split becomes the
+// working copy commit for all workspaces whose working copy commit was the
+// target of the split. This test does a split where the target commit is the
+// working copy commit for two different workspaces.
+#[test]
+fn test_split_with_multiple_workspaces_same_working_copy() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "main"]);
+    let main_path = test_env.env_root().join("main");
+    let secondary_path = test_env.env_root().join("secondary");
+
+    test_env.jj_cmd_ok(&main_path, &["desc", "-m", "first-commit"]);
+    std::fs::write(main_path.join("file1"), "foo").unwrap();
+    std::fs::write(main_path.join("file2"), "foo").unwrap();
+
+    // Create the second workspace and change its working copy commit to match
+    // the default workspace.
+    test_env.jj_cmd_ok(
+        &main_path,
+        &["workspace", "add", "--name", "second", "../secondary"],
+    );
+    // Change the working copy in the second workspace.
+    test_env.jj_cmd_ok(
+        &secondary_path,
+        &["edit", "-r", "description(first-commit)"],
+    );
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: qpvuntsm d12cb1c0 first-commit
+    second: qpvuntsm d12cb1c0 first-commit
+    "###);
+    // Check the working-copy commit in each workspace in the log output. The "@"
+    // node in the graph indicates the current workspace's working-copy commit.
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  qpvuntsmwlqt default@ second@ first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  qpvuntsmwlqt default@ second@ first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+
+    // Do the split in the default workspace.
+    std::fs::write(
+        test_env.set_up_fake_editor(),
+        ["", "next invocation\n", "write\nsecond-commit"].join("\0"),
+    )
+    .unwrap();
+    test_env.jj_cmd_ok(&main_path, &["split", "file2"]);
+    // The working copy for both workspaces will be the second split commit.
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: vruxwmqv 97490ec0 second-commit
+    second: vruxwmqv 97490ec0 second-commit
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  vruxwmqvtpmx default@ second@ second-commit
+    ○  qpvuntsmwlqt first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  vruxwmqvtpmx default@ second@ second-commit
+    ○  qpvuntsmwlqt first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
```

Comment:
nit: this seems redundant with the check just above unless I'm missing something

Reply:
Yeah, it's redundant. I think it would be better to remove the `jj workspace list` call and just leave a single log call since the log call shows the commit graph. Having the graph makes the test easier to follow, at least IMO.